editors/code/src/extension.ts

```diff
@@ -147,6 +148,21 @@ export function activate(context: vscode.ExtensionContext) {
 
     // Start the language server, finally!
     startServer();
+
+    if (Server.config.displayInlayHints) {
+        const hintsUpdater = new HintsUpdater();
+        hintsUpdater.loadHints(vscode.window.activeTextEditor).then(() => {
+            vscode.window.onDidChangeActiveTextEditor(editor =>
```

Comment:
I think listeners should be explicitly registered for disposal, using `disposeOnDeactivation`

Reply:
Should we also do this to the 

```
vscode.workspace.onDidChangeTextDocument(
        events.changeTextDocument.createHandler(syntaxTreeContentProvider),
        null,
        context.subscriptions
    );
```
?

Reply:
Done for all new ones.